### PR TITLE
Make automapper config single instance

### PR DESCRIPTION
**Pull request by Gurnaik-Randhawa**
Make automapper config single instance
Make Sitefinity dynamic module manager single instance

### CODE REVIEW

**Review by keenesolutions — APPROVED** [review]
Looks good.

File: DFC.Digital/DFC.Digital.Web.Core/Config/WebCoreAutofacConfig.cs [hunk 1]
```diff
@@ -96,7 +96,7 @@ private static void RegisterAutomapper(ContainerBuilder builder, IEnumerable<Ass
                     cfg.AddProfile(profile);
                 }
             })).AsSelf().InstancePerLifetimeScope();
-            builder.Register(ctx => ctx.Resolve<MapperConfiguration>().CreateMapper()).As<IMapper>().InstancePerLifetimeScope();
+            builder.Register(ctx => ctx.Resolve<MapperConfiguration>().CreateMapper()).As<IMapper>().SingleInstance();
```
Comment (keenesolutions) [inline DFC.Digital/DFC.Digital.Web.Core/Config/WebCoreAutofacConfig.cs:99]: This should remain as InstancePerLifetimeScope().

File: DFC.Digital/DFC.Digital.Web.Core/Config/WebCoreAutofacConfig.cs [hunk 2]
```diff
@@ -96,7 +96,7 @@ private static void RegisterAutomapper(ContainerBuilder builder, IEnumerable<Ass
                     cfg.AddProfile(profile);
                 }
             })).AsSelf().InstancePerLifetimeScope();
```
Comment (keenesolutions) [inline DFC.Digital/DFC.Digital.Web.Core/Config/WebCoreAutofacConfig.cs:98]: This needs changing from This should remain as **InstancePerLifetimeScope()** to **SingleInstance()**

Comment (keenesolutions) [reply DFC.Digital/DFC.Digital.Web.Core/Config/WebCoreAutofacConfig.cs:98]: This is the change that would have had the biggest impact in testing as we are no longer configuring automapper for each instance requested.

File: DFC.Digital/DFC.Digital.Web.Sitefinity.JobProfileModule/Mvc/Controllers/PreSearchFiltersController.cs [hunk 3]
```diff
@@ -102,12 +102,15 @@ public class PreSearchFiltersController : BaseDfcController
...
-        [DisplayName("Should show number of profile matched banner")]
+        [DisplayName("Number of profle matching on each page message")]
         public string NumberOfMatchesMessage { get; set; } = "We have found {0} career matches based on your selection.";
 
         [DisplayName("Select Message")]
         public string SelectMessage { get; set; } = @"<div class=""govuk-hint"" id=""qualifications-hint"">Select all that apply.</div>";
 
+        [DisplayName("Use Page Profile Count")]
+        public bool UsePageProfileCount { get; set; } = true;
+
```
Comment (keenesolutions) [inline DFC.Digital/DFC.Digital.Web.Sitefinity.JobProfileModule/Mvc/Controllers/PreSearchFiltersController.cs:113]: Does this now affect current unit/acceptance tests or do we need to add some to prove that wehn this is false we con't make a call to GetNumberOfMatches? Happy for this to be done later if this is urgent for perf testing in pre-prod.

**Review by keenesolutions — CHANGES_REQUESTED** [review]
Need to make a few updates to autofacConfig.

File: DFC.Digital/DFC.Digital.Web.Sitefinity.JobProfileModule/Mvc/Controllers/PreSearchFiltersController.cs (hunk 3 above)
Comment (Gurnaik-Randhawa) [reply DFC.Digital/DFC.Digital.Web.Sitefinity.JobProfileModule/Mvc/Controllers/PreSearchFiltersController.cs:113]: Should be done now

**Review by keenesolutions — APPROVED** [review]
Looks good.

**Pull request merged**